db/collab.go

```diff
@@ -164,8 +206,25 @@ func (d *DB) JoinCollab(c echo.Context) error {
 				break
 			}
 
-			if broadcastErr := session.Broadcast(msg); broadcastErr != nil {
-				break
+			switch msg.Type {
+			case "READ":
+				if msg.Author == session.Teacher {
+					if studentSendTo, ok := session.SendTo[msg.Target]; ok {
+						studentSendTo[uid] = true
+					} else {
+						if err := session.BroadcastError(uid, "Student does not exist"); err != nil {
+							break
+						}
+					}
+				} else {
+					if err := session.BroadcastError(uid, "Teacher permission required to request access to student"); err != nil {
+						break
+					}
+				}
+			default:
+				if err := session.BroadcastToSet(msg, session.SendTo[uid]); err != nil {
+					break
+				}
```

Comment:
The rightward pressure here is too great. I think this packet management should be abstracted to a member function of our `Session` type. Alternatively, the introduction of our extension of the `Conn` type could solve this issue, too.

Reply:
Can you elaborate on what exactly should be abstracted here? Sorry, not entirely sure what you're referring to